### ISSUE TITLE
Xml documentation file not created by MSBuild

### ISSUE THREAD

**Issue by davkean** [dotnet/sdk#1010]
_From @gmsDave on March 20, 2017 3:35_

I have an asp.net core web api project that I have recently converted from project.json to .csproj.  I am using Swashbuckle to generate a Swagger UI which relies on the generated xml documentation file.  In project.json, this was created using and "xmlDoc": "true" element in the "buildOptions". In .csproj, it looks like this was converted to a GenerateDocumentationFile element in a PropertyGroup xml.  All seems well - when I build and run the project through VS2017, the file is created and the swagger UI works as expected:

![image](https://cloud.githubusercontent.com/assets/18301427/24087802/236b326a-0ce9-11e7-8070-695d5834730d.png)

Taking a look at the bin folder and the expected .xml file exists

![capture4](https://cloud.githubusercontent.com/assets/18301427/24087834/6d4cf7a6-0ce9-11e7-8074-ce547259f64c.jpg)

When I run a build, either through TFS (on-prem) using the Visual Studio Build task:

![image](https://cloud.githubusercontent.com/assets/18301427/24087847/988c921e-0ce9-11e7-90ff-ebc95dd345d3.png)

or from a command line:

![capture2](https://cloud.githubusercontent.com/assets/18301427/24087959/9aead0d8-0cea-11e7-8198-a94dbe85b9f7.JPG)

the xml doc file isn't generated:

![capture3](https://cloud.githubusercontent.com/assets/18301427/24088089/ec79d6fa-0ceb-11e7-9fff-84e98f1f35ab.png)

I'm wondering if this is a bug or if there is a switch or option that I need to enable. Here is a snippet of my .csproj file:

![capture1](https://cloud.githubusercontent.com/assets/18301427/24088008/2d42896c-0ceb-11e7-81ae-1b0fd0ca713f.JPG)

Note that I am targeting the full .net framework - not sure if that matters.  Here is the build log from the TFS task: 

[6_Build solution CIMS.NETCIMS.sln.txt](https://github.com/dotnet/roslyn-project-system/files/853882/6_Build.solution.CIMS.NETCIMS.sln.txt)

and here is from the manual one I did from the command line:

[manual_build.txt](https://github.com/dotnet/roslyn-project-system/files/853883/manual_build.txt)

_Copied from original issue: dotnet/roslyn-project-system#1793_

**Comment by davkean** [comment]
Ah, thought this was not occurring when building though VS, however, it is. I'll move this to the SDK.

**Comment by DaveSlinn** [comment]
Duplicate of https://github.com/dotnet/sdk/issues/795

**Comment by livarcocc** [comment]
Yep, I believe this is a dup of #795.

**Comment by beinguam** [comment]
I am having the exact same issue generating the XML documentation file using TFS when it builds the solution.  Any fix?

**Comment by dasMulli** [comment]
@beinguam are you using the 2.0.0 SDK? This version has the code to publish the documentation file.
If that doesn't work for you, can you share more detail on your setup / csproj file?

**Comment by beinguam** [comment]
It is actually a WebAPI on the full .NET framework (4.5.2) not .NET Core.   What additional details do you need?

**Comment by dasMulli** [comment]
An ASP.NET Core "web api" project or classic web api? if classic, that's not an issue with any of the components in this repo. (microsoft/msbuild maybe but I don't know)

**Comment by beinguam** [comment]
Nope , it is a classic web api, my mistake.  let me look around

**Comment by SandeepApsingekar** [comment]
I have the similar issue, was this issue resolved?
Any help would be appreciated. 
Thanks

**Comment by forX01** [comment]
got the same issue, .net api classic + azure build. (I think the standard VS publish do it too)

**Comment by Jasonlhy** [comment]
I had this problem as well and I spent a few hours to understand why. The problem is xml extension files are excluded in the publish stage.

You can check the target file at `C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\VisualStudio\v16.0\Web\Microsoft.Web.Publishing.targets`

```xml
    <!-- This will control whether XML files from ResolveAssemblyReference is published, default is true (to exclude) -->
    <ExcludeXmlAssemblyFiles Condition=" '$(ExcludeXmlAssemblyFiles)'=='' ">true</ExcludeXmlAssemblyFiles>
```

Solution:
Make this to false to copy all xml file or add a custom task after publish to copy the required files.
```xml
<PropertyGroup>
    <ExcludeXmlAssemblyFiles>false</ExcludeXmlAssemblyFiles>
  </PropertyGroup>
```

Last but not least:
dotnet sdk publish and MSBuild publish use different target settings.
This only work for classic .NET framework
.NET SDK have different property settings

**Comment by dasMulli** [comment]
Yes this issue is purely for ASP.NET Core and the .NET SDK, you can file requests for other VS components at https://developercommunity.visualstudio.com/

**Comment by KvnSearle** [comment]
I'm having this issue with a .Net6 application developed using MSVS2022.

We upgraded the application from .Net Core 3.1 to .net 6 recently and I am wondering if this has caused the change.

If I publish the files locally using visual studio the xml file is created.

However it's missing from the zip file artefact created by the devops build pipeline.

I added the following to the csproj file but this has not helped
    <GenerateDocumentationFile>true</GenerateDocumentationFile>

Anyone got any ideas of how to resolve this?

**Comment by baronfel** [comment]
@KvnSearle you may want to get an https://aka.ms/binlog of your DevOps build pipeline and compare it to the binlog of your local publish using Visual Studio. Without that raw data, we can't really dig into the _why_ with any accuracy.

**Comment by KvnSearle** [comment]
@baronfel, thanks for the feedback.

I found a solution for this which was more prosaic..

In the build pipeline I set the build configuration to be 'Release' and the build platform to be 'Any CPU' and the xml files were generated.

I originally created the YAML file for the build pipeline by dynamically generating the YAML from an existing classic build pipeline.

It did not include either the build configuration or build platform setting so I took it that it wasn't needed.

The API had been working fine (as far as I am aware) it was only the need to view it in swagger that highlighted the problem.
